File: lib/layer2_core/dark/e_on_tertiary_fixed_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EOnTertiaryFixedDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd39;
      case 'standard': return cstd39;
      case 'blue': return cbd39;
      case 'green': return cgd39;
      case 'red': return crd39;
      case 'nothing': return cnod39;
      default: return cstd39;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EOnTertiaryFixedDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnTertiaryFixedDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd39
            AppTheme.STANDARD -> R.color.cstd39
            AppTheme.BLUE -> R.color.cbd39
            AppTheme.GREEN -> R.color.cgd39
            AppTheme.RED -> R.color.crd39
            AppTheme.NOTHING -> R.color.cnod39
        }
    }
}
